operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/internal/CustomResourceEventSource.java

```diff
@@ -155,7 +180,8 @@ public void onClose(WatcherException e) {
     if (e.isHttpGone()) {
       log.warn("Received error for watch, will try to reconnect.", e);
       try {
-        registerWatch();
+        close();
```

Comment:
Should we really close all the watches? I guess there's no other option since we can't really know which watcher caused the exception…

Reply:
yes, the issue was that in case of an error on of a single watcher, calling `registerWatch` would re-register all the watcher, potentially you could end up in duplicated watchers.

Reply:
That was another thing I wondered about (using a List for the watchers as opposed to a Set but I'm not sure we have  consistent `equals/hashCode` implementations for the `Watch` implementations)…

Reply:
yes we need a way to improve how watchers are handled, I may try to have a look once this is done